src/components/common/S3Upload/RenderS3Upload.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { postNewUpload, postSubmissionPage } from '../../../api';
+import { UploadOutlined } from '@ant-design/icons';
+import { Button, Upload, message } from 'antd';
+import { useState } from 'react';
+
+export default function S3UploadButton() {
+  const [file, setFile] = useState([]);
+  const [uploading, setUploading] = useState(false);
+
+  const fileSelected = fileName => {
+    const selectedFile = fileName.file;
+    setFile(selectedFile);
+    console.log('file', selectedFile);
+  };
+
+  const handleUpload = e => {
+    e.preventDefault();
+    setUploading(true);
+    postNewUpload(file)
+      .then(res => {
+        console.log('response from 1st .then', res);
+        postSubmissionPage({ url: res })
+          .then(res => {
+            console.log('response from uploading url to DB', res);
+          })
+          .catch(err => {
+            console.log('error from uploading url to DB', err);
+          });
+      })
+      .then(() => {
+        setFile([]);
```

Comment:
Should these lines be on line 25 in that nested call promise for postSubmissionPage()

Reply:
Yes that extra .then convolutes the code.